python/tk_multi_loader/dialog.py

```diff
@@ -1265,6 +1280,94 @@ def _load_entity_presets(self):
         # data has properly arrived in the model.
         self._on_home_clicked()
 
+    def _setup_hierarchy_model(self, app, root):
+        """
+        Create the model and proxy model required by a hierarchy type configuration setting.
+
+        :param app: :class:`Application`, :class:`Engine` or :class:`Framework` bundle instance
+                    associated with the loader.
+        :param root: The path to the root of the Shotgun hierarchy to display.
+        :return: Created `(model, proxy model)`.
+        """
+
+        # Resolve any magic tokens in the root.
+        if app.context.project:
+            project_id = app.context.project.get("id")
+            root = root.replace("{context.project.id}", str(project_id))
+        else:
+            root = None
+
+        # Construct the hierarchy model.
+        model = SgHierarchyModel(self, bg_task_manager=self._task_manager)
```

Comment:
i think i'd prefer if we would refactor this into a class, like it's done for the sg model: https://github.com/shotgunsoftware/tk-multi-loader2/blob/master/python/tk_multi_loader/model_entity.py#L54